### PR TITLE
Rename files, structs and plugin registered names

### PR DESCRIPTION
**Pull request by ashish-amarnath**
Signed-off-by: Ashish Amarnath ashisham@vmware.com

In this change:

- rename files to follow `<type>_action.go` and `<type>_action_test.go` naming scheme in internal/backup and internal/restore directories
- rename plugin structs to follow `<Type>BackupItemAction` and `<Type>RestoreItemAction` naming scheme
- follow `velero.io/csi-<type>-backuppe`r and `velero.io/csi-type-restorer` convention to register plugins

Was unable to revive PR #37  opening a new one in its place with the changes redone

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -27,19 +27,19 @@ import (
...
-		RegisterRestoreItemAction("velero.io/volumesnapshotcontent-restorer", newVSCRestorer).
-		RegisterRestoreItemAction("velero.io/volumesnapshot-restorer", newVSRestorer).
+		RegisterBackupItemAction("velero.io/csi-pvc-backupper", newPVCBackupItemAction).
+		RegisterBackupItemAction("velero.io/csi-volumesnapshot-backupper", newVolumeSnapshotBackupItemAction).
+		RegisterBackupItemAction("velero.io/csi-volumesnapshotclass-backupper", newVolumesnapshotClassBackupItemAction).
+		RegisterBackupItemAction("velero.io/csi-volumesnapshotcontent-backupper", newVolumeSnapContentBackupItemAction).
+		RegisterRestoreItemAction("velero.io/csi-pvc-restorer", newPVCRestoreItemAction).
+		RegisterRestoreItemAction("velero.io/csi-volumesnapshotclass-restorer", newVolSnapClassRestorer).
+		RegisterRestoreItemAction("velero.io/csi-volumesnapshotcontent-restorer", newVSCRestorer).
+		RegisterRestoreItemAction("velero.io/csi-volumesnapshot-restorer", newVolumeSnapshotRestoreItemAction).
```
Comment (skriss) [inline main.go:37]: minor nit but it'd be nice to move L37 up to just above L35, so the plugins are ordered the same for backup & restore.

File: main.go [hunk 2]
```diff
@@ -54,16 +54,16 @@ func newVolumeSnapContentBackupItemAction(logger logrus.FieldLogger) (interface{
 	return &backup.VolumeSnapshotContentBackupItemAction{Log: logger}, nil
 }
 
-func newCSIRestorer(logger logrus.FieldLogger) (interface{}, error) {
-	return &restore.CSIRestorer{Log: logger}, nil
+func newPVCRestoreItemAction(logger logrus.FieldLogger) (interface{}, error) {
+	return &restore.PVCRestoreItemAction{Log: logger}, nil
 }
 
 func newVSCRestorer(logger logrus.FieldLogger) (interface{}, error) {
```
Comment (skriss) [inline main.go:61]: Let's rename this helper func as well to be consistent

File: main.go [hunk 3]
```diff
@@ -54,16 +54,16 @@ func newVolumeSnapContentBackupItemAction(logger logrus.FieldLogger) (interface{
...
 	return &restore.VSCRestorer{Log: logger}, nil
 }
 
-func newVSRestorer(logger logrus.FieldLogger) (interface{}, error) {
-	return &restore.VSRestorer{Log: logger}, nil
+func newVolumeSnapshotRestoreItemAction(logger logrus.FieldLogger) (interface{}, error) {
+	return &restore.VolumeSnapshotRestoreItemAction{Log: logger}, nil
 }
 
 func newVolSnapClassRestorer(logger logrus.FieldLogger) (interface{}, error) {
```
Comment (skriss) [inline main.go:69]: let's rename this helper func as well to be consistent

**Comment by nrb** [comment]
> I'd change internal/restore/volumesnapshotcontents_action.go to be singular, i.e. internal/restore/volumesnapshotcontent_action.go since I think that's the actual type name and consistent with other names.

I thought it was plural, but [you are correct](https://github.com/kubernetes-csi/external-snapshotter/blob/master/pkg/apis/volumesnapshot/v1beta1/types.go#L233)

**Pull request merged**